microprofile.cpp

```diff
@@ -12623,7 +12635,7 @@ bool MicroProfileHashTableSet(MicroProfileHashTable* pTable, uint64_t Key, uintp
 	while(1)
 	{
 		const uint32_t nLim = pTable->nLim;
-		uint32_t B = H % pTable->nAllocated;
+		uint32_t B = static_cast<uint32_t>(H) % pTable->nAllocated;
```

Comment:
Triggers SIGFPE, because H is a uint64 and will overflow to get it to fit within the modulo operation. The cast makes the existing behavior explicit/not-error-y, but I have not checked to see what that means for the hash table generally

Reply:
-Can you move this to a seperate 
-use C style casts.
-I count 3 instances of the bug, it seems you only have two of them. Can you add the last one. (one in each of Get/Set/Remove)